ada/ast.py

```diff
@@ -14000,6 +14051,25 @@ class Identifier(BaseId):
         )
     )
 
+    @langkit_property()
+    def is_constant():
+        return Or(
+            Entity.referenced_decl.is_constant_object,
+
+            # An instance of a protected unit is constant within
+            # a function body of the corresponding protected type.
+            Entity.parents.find(
+                lambda n: n.cast(T.SubpBody)._.subp_spec
+                .subp_kind.is_a(SubpKind.alt_function)
```

Comment:
You're not checking that the function is defined inside a protected. I imagine that still works, because if you have visibility  on those protected variables, you are necessarily in a protected object, but if that was your reasoning this should be explained in a comment IMO